clients/tabby-chat-panel/src/index.ts

```diff
@@ -22,6 +22,15 @@ export interface FetcherOptions {
 
 export interface InitRequest {
   fetcherOptions: FetcherOptions
+  focusKey?: FocusKeybinding
+}
+
+export interface FocusKeybinding {
+  key: string
+  ctrlKey: boolean
+  metaKey: boolean
+  shiftKey: boolean
+  altKey: boolean
```

Comment:
Can we bubble all uncosumed key events to the VS Code webview and process them there? This way, we wouldn't have to pass the keybinding configurations to the chat panel implementation.   
It would also help clean up the handling of copy/paste key events, which are only necessary in the VS Code environment. Additionally, if we need to support more keybindings, the API can remain stable. For example, if we want to add compatibility to use `ctrl+shift+p` to invoke the command palette when the chat panel is in focus.

Reply:
When passing additional data, simply bubbling keyboard events is insufficient; it necessitates an extra server API call. (keyboard event -&gt; vscode call `ServerAPI.getData`, which seems cumbersome).

Conversely, handling events within the webview allows proactive data passing. (No data is passed in `ClientAPI.focusEditor`, but it could be similar.)

Reply:
Keyboard events should never contain additional data.  
If you are referring to the `onCopy` api requiring additional data, it is used when clicking the tool icon on the code block; it is different from pressing `ctrl/cmd + c`.

Reply:
offline resolved